packages/devtools_app/lib/src/screens/memory/shared/heap/heap.dart

```diff
@@ -10,123 +10,113 @@ class AdaptedHeap {
 
   final AdaptedHeapData data;
 
-  late final HeapStatistics stats = _heapStatistics(data);
+  late final SingleHeapClasses classes = _heapStatistics(data);
 
-  static HeapStatistics _heapStatistics(AdaptedHeapData data) {
-    final result = <String, HeapClassStatistics>{};
+  static SingleHeapClasses _heapStatistics(AdaptedHeapData data) {
+    final result = <HeapClassName, SingleClassStats>{};
     if (!data.isSpanningTreeBuilt) buildSpanningTree(data);
 
     for (var i in Iterable.generate(data.objects.length)) {
       final object = data.objects[i];
-      final heapClass = object.heapClass;
+      final className = object.heapClass;
 
       // We do not show objects that will be garbage collected soon or are
       // native.
-      if (object.retainedSize == null || heapClass.isSentinel) continue;
+      if (object.retainedSize == null || className.isSentinel) continue;
 
-      final fullName = heapClass.fullName;
-
-      final stats =
-          result.putIfAbsent(fullName, () => HeapClassStatistics(heapClass));
-      stats.countInstance(data, i);
+      final singleHeapClass =
+          result.putIfAbsent(className, () => SingleClassStats(className));
+      singleHeapClass.countInstance(data, i);
     }
 
-    return HeapStatistics(result)..seal();
+    return SingleHeapClasses(result)..seal();
   }
 }
 
-class HeapStatistics {
-  HeapStatistics(this.statsByClassName);
+abstract class HeapClasses with Sealable {
+  // HeapClass? classByName(HeapClassName? name);
+}
```

Comment:
should this class be removed

Reply:
it is used in SnapshotInstanceItem.classesToShow